lib/takeaway.rb

```diff
@@ -0,0 +1,89 @@
+require_relative 'send_sms'
+require_relative 'restaurant'
+
+class Takeaway
+
+  attr_reader :menu, :order
+
+  def initialize(restaurant_klass=Restaurant)
+    @restaurant = restaurant_klass
+    @order = Hash.new(0)
+  end
+
+  def available
+    "Available cuisines: #{@restaurant::CUISINES.keys}"
+  end
+
+  def new_order(cuisine=:Italian)
+    @menu = @restaurant.new(cuisine).menu
+    "A new order has been opened. Please read the menu"
+  end
+
+  def add(item, quantity=1)
+    if menu.include?(item)
+      order[item] += quantity
+      "#{item} x#{quantity} has been added to your order"
+    else
+      "Sorry, your selection is unavailable"
+    end
+  end
+
+  def delete(item, quantity=1)
+    if order.include?(item)
+      order[item] -= quantity
+      "#{item} x#{quantity} has been deleted from your order"
+    else
+      "Sorry, your order does not contain that item"
+    end
+  end
+
+  def order_summary
+    summary = order.map { |item,quantity| "#{item} x#{quantity} = #{gbp(menu[item]*quantity)}" }.join(', ')
```

Comment:
Metrics/LineLength: Line is too long. [107/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)